src/net_processing.cpp

```diff
@@ -2625,7 +2625,11 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             pfrom.fDisconnect = true;
             return;
         }
-        pfrom.m_wants_addrv2 = true;
+
+        if (SetupAddressRelay(pfrom)) {
```

Comment:
I don't think there's any need to `SetupAddressRelay` here, and doing so would cause you to continue choosing inbounds that are treating you as block-relay-only instead of ignoring them as a black hole.

Reply:
I don't think it matters much either way. The only peers this affects are inbound block-relay-only connections from v0.21 peers, which (incorrectly) send `sendaddrv2` to us. Inbound block-relay-only peers on v22 won't send `sendaddrv2` because of the change in commit _Stop sending SENDADDRV2 message to block relay only peers_.

Reply:
ergh, its annoying that v0.21 peers will send us the `sendaddrv2` message, but I'm not sure if this warrants removing this conditional. some thoughts:
- in terms of protocol & logical coherence, I'd expect `sendaddrv2` to indicate desire that the node wants to participate in addr relay
- in this patch, calling `SetupAddressRelay` for the node mimics the behavior of current master. so I think of calling as the default (rather than the other way around)
- in terms of safety, it seems better to have a blackhole than exclude a peer from address relay.   I think... a blackhole means an address, or set of addresses will not propagate as well this time, but excluding a peer from address relay (esp if programmatically / many peers exclude) means that peer could be at a higher risk of being eclipsed? 

I thought I was on the fence, but writing out this reasoning makes me realize I think I'd like to leave as is.

thoughts?

Reply:
Leaving this as is - I find it most logically consistent for `sendaddrv2` to indicate "interested in addresses". Opened #22245 to change the behavior for inbound block-relay-only v22.0 peers.